### ISSUE TITLE
ottr::export fails when pdf can’t be generated

### ISSUE THREAD

**Issue by chrispyles** [ucbds-infra/otter-grader#766]
look into both why the pdf can’t be generated and why the zip export fails

Thanks @… here it is https://github.com/dbroockman/PS3-SP24-Public/blob/main/w2/w2a1/student/Week2_Activity1.ipynb

[Slack Message](https://otter-grader.slack.com/archives/CT9QJK4PL/p1705878030805889?thread_ts=1705872440.511379&cid=CT9QJK4PL)

**Comment by dbroockman** [comment]
Looks like it's not just a PDF issue, I get part of the error on Berkeley R Datahub when setting `pdf = FALSE` too:
![Screenshot 2024-01-21 at 9 06 21 PM](https://github.com/ucbds-infra/otter-grader/assets/6677808/c11907a8-54fb-4a12-a43e-fa1f9d009595)

**Comment by dbroockman** [comment]
And actually, I even get an error without the `force_save = TRUE` flag.
<img width="631" alt="Screenshot 2024-01-21 at 9 31 02 PM" src="https://github.com/ucbds-infra/otter-grader/assets/6677808/8449931e-1742-4720-b743-ab0c269390d0">

**Comment by dbroockman** [comment]
Upon further investigation, it's possible at least some of this might be tied to generic Berkeley datahub issues. https://ds-modules.slack.com/archives/C3ESHBX2N/p1705901646319269

**Comment by chrispyles** [comment]
> Looks like it's not just a PDF issue, I get part of the error on Berkeley R Datahub when setting `pdf = FALSE` too: ![Screenshot 2024-01-21 at 9 06 21 PM](https://private-user-images.githubusercontent.com/6677808/298467446-c11907a8-54fb-4a12-a43e-fa1f9d009595.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MDMxMjUsIm5iZiI6MTcwNTkwMjgyNSwicGF0aCI6Ii82Njc3ODA4LzI5ODQ2NzQ0Ni1jMTE5MDdhOC01NGZiLTRhMTItYTQzZS1mYTFmOWQwMDk1OTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjJUMDU1MzQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTQzNDE0ZTA1MTE5ZTM5NzNhMzM5OTJmNmJhY2YyNDIzMzI3YTcwMjcwYzZlNWUwMjA3NzkyYTlmZjBiMTdlNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.GOYCmWXZWYeDzMeIES8QaAvR06VLS5MsgXbSFjZbgdI)

The warning you're seeing might be a separate issue -- I'm not sure why force save isn't working. But from the line below the warning, you can see that the zip file _was_ generated, so the bug in this issue is only occurring when `pdf = TRUE`.

**Comment by dbroockman** [comment]
Thanks. With the Datahub issues resolved, looks like the issue persists with force save set to TRUE but the function works without either the force save or PDF flags.

**Comment by chrispyles** [comment]
Sorry, I'm confused now. Can you send a screenshot of running with both pdf and force_save set to true now that the hub issues are resolved?

**Comment by dbroockman** [comment]
Summary:

- `ottr::export("Week2_Activity1.ipynb")` works
- `ottr::export("Week2_Activity1.ipynb", force_save = TRUE)` error: 
<img width="1321" alt="Screenshot 2024-01-22 at 9 17 40 AM" src="https://github.com/ucbds-infra/otter-grader/assets/6677808/742e840d-ab81-47d3-9cdb-fb1ca0b5085b">
- `ottr::export("Week2_Activity1.ipynb", pdf = TRUE)` error: 
<img width="1289" alt="Screenshot 2024-01-22 at 9 18 08 AM" src="https://github.com/ucbds-infra/otter-grader/assets/6677808/30592878-6cf2-45e5-8ca9-c1d392fb166d">

**Comment by chrispyles** [comment]
What you're seeing in `ottr::export("Week2_Activity1.ipynb", force_save = TRUE)` isn't an error but a warning; the zip is still being generated. Not sure why force save isn't working but I'll open a separate issue for that.

For `ottr::export("Week2_Activity1.ipynb", pdf = TRUE)`, can you open a terminal on the hub, `cd` into the directory with the notebook, and run `jupyter nbconvert --to=pdf Week2_Activity1.ipynb` and set the output?

**Comment by chrispyles** [comment]
> For ottr::export("Week2_Activity1.ipynb", pdf = TRUE), can you open a terminal on the hub, cd into the directory with the notebook, and run jupyter nbconvert --to=pdf Week2_Activity1.ipynb and send the output?

@dbroockman, any update on this?

**Comment by dbroockman** [comment]
```jovyan@jupyter-dbroockman:~/PS3-SP24-Public/w2/w2a1/student$ jupyter nbconvert --to=pdf Week2_Activity1.ipynb
Traceback (most recent call last):
  File "/srv/conda/bin/jupyter-nbconvert", line 6, in <module>
    from nbconvert.nbconvertapp import main
  File "/srv/conda/lib/python3.9/site-packages/nbconvert/nbconvertapp.py", line 193, in <module>
    class NbConvertApp(JupyterApp):
  File "/srv/conda/lib/python3.9/site-packages/nbconvert/nbconvertapp.py", line 252, in NbConvertApp
    Options include {get_export_names()}.
  File "/srv/conda/lib/python3.9/site-packages/nbconvert/exporters/base.py", line 145, in get_export_names
    e = get_exporter(exporter_name)(config=config)
  File "/srv/conda/lib/python3.9/site-packages/nbconvert/exporters/base.py", line 106, in get_exporter
    exporter = items[0].load()
  File "/srv/conda/lib/python3.9/site-packages/importlib_metadata/__init__.py", line 184, in load
    module = import_module(match.group('module'))
  File "/srv/conda/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/srv/conda/lib/python3.9/site-packages/jupyter_contrib_nbextensions/__init__.py", line 5, in <module>
    import jupyter_nbextensions_configurator
  File "/srv/conda/lib/python3.9/site-packages/jupyter_nbextensions_configurator/__init__.py", line 18, in <module>
    from notebook.base.handlers import APIHandler, IPythonHandler
ModuleNotFoundError: No module named 'notebook.base'
```

**Comment by chrispyles** [comment]
Looks like the hub has some incompatible package versions; I'm guessing it's an outdated version of `jupyter_nbextensions_configurator`. I'd open an issue on the datahub repo that this is preventing nbconvert from being able to export notebooks as PDFs.

From [this](https://stackoverflow.com/questions/76893872/modulenotfounderror-no-module-named-notebook-base-when-installing-nbextension) it looks like notebook v7 (which is what the hub is using) isn't compatible with notebook extensions? I'm not 100% sure.

**Comment by dbroockman** [comment]
Thank you, bug opened here: https://github.com/datahub-project/datahub/issues/9732

**Comment by chrispyles** [comment]
I think that's the wrong repo. Berkeley datahub (afaik) is run from https://github.com/berkeley-dsep-infra/datahub

**Comment by dbroockman** [comment]
Thanks for the flag.

**Comment by shaneknapp** [comment]
> Looks like the hub has some incompatible package versions; I'm guessing it's an outdated version of `jupyter_nbextensions_configurator`. I'd open an issue on the datahub repo that this is preventing nbconvert from being able to export notebooks as PDFs.
> 
> From [this](https://stackoverflow.com/questions/76893872/modulenotfounderror-no-module-named-notebook-base-when-installing-nbextension) it looks like notebook v7 (which is what the hub is using) isn't compatible with notebook extensions? I'm not 100% sure.

yep, this is 100% the case:  notebook v7 isn't compatible with notebook v6 extensions.

**Comment by chrispyles** [comment]
I've opened https://github.com/ucbds-infra/ottr/pull/21 to continue exporting the submission zip if a PDF can't be generated, which should be the only change required on otter's end for this issue. It will be included in v1.5.0 of ottr, which will come out once I've fixed #767 as well.
